### ISSUE TITLE
[Bug] Loading Overlay closes too soon with onApprove() callback

### ISSUE THREAD

**Issue by gregjopa** [paypal/paypal-checkout-components#1486]
### 🐞 Describe the Bug

The Loading Overlay disappears as soon as the transaction is done and doesn't wait for the `onApprove` callback to complete.

### 🔬 Minimal Reproduction

This problem can be reproduced by adding an additional async call to the `onApprove` callback like so:

```js
    onApprove: function (data, actions) {
      return actions.order
        .capture()
        .then(sleep(5000))
        .then(() => {
          alert("transaction complete!");
        });
    }
```

Full Demo: https://codepen.io/gregjopa/pen/VwKvYqY

### 😕 Actual Behavior

The Loading Overlay disappears before the `onApprove` Promise resolves. In the above demo the Loading Overlay disappears before the sleep operation finishes. This can confuse the buyer and lead them to believe that their payment did not go through.

### 🤔 Expected Behavior

You would expect the Loading Overlay to disappear after the `onApprove` Promise resolves.

**Comment by mnicpt** [comment]
@gregjopa For which flow does this occur?

**Comment by yocmen** [comment]
Hi @mnicpt in my case, I made all the processes with Paypal then on Approve I send the status/data with a fetch to my backend-server to register the sale and all the process to attach the product to the user. This is part of the flow, but as the overlay is closed it feels like it finished even when is not done yet, I handled that right now by adding a new loading in the back of the Paypal overlay, but it feels "ugly", and I need to wait for this results to then redirect to the user to the thanks u page with some data like order number(from my final fetch to the backend) (I use full gatsbyjs/static pages and APIS for my backend) you can go to vtutor.com and see my working flow on the checkout page, really thanks for this initiative wich will improve the Paypal experience for developers and end users.

**Comment by colinrim** [comment]
same issue here, also experienced with https://github.com/Luehang/react-paypal-button-v2.
confusing for buyer as state of blank freezes for several seconds before app takes over again.

**Comment by mnicpt** [comment]
@yocmen @colinrim We have this on our backlog to better detect when to close the overlay for a better UX.  Will let you know when this is complete.

**Comment by leogedler** [comment]
Opened PR with a solution for this problem: https://github.com/paypal/paypal-smart-payment-buttons/pull/170

**Comment by leogedler** [comment]
A solution for this problem was released with the JS SDK version `5.0.236`. 

Now the overlay/popup will remain displayed till all promises inside the `onApprove` callback get resolved.

**Comment by gregjopa** [comment]
Thanks for the update @leogedler. Awesome work adding this feature! 💯 🥇 👍
